crates/driver/src/domain/mempools.rs

```diff
@@ -63,6 +74,94 @@ impl Mempools {
             RevertProtection::Enabled
         }
     }
+
+    async fn submit(
+        &self,
+        mempool: &infra::mempool::Inner,
+        solver: &Solver,
+        settlement: &Settlement,
+    ) -> Result<eth::TxId, Error> {
+        let eth = &self.1;
+        let deadline = tokio::time::Instant::now() + mempool.config().max_confirm_time;
+        let tx = eth::Tx {
+            // boundary.tx() does not populate the access list
+            access_list: settlement.access_list.clone(),
+            ..settlement.boundary.tx(
+                settlement.auction_id,
+                self.1.contracts().settlement(),
+                competition::solution::settlement::Internalization::Enable,
+            )
+        };
+        let hash = mempool.submit(tx.clone(), settlement.gas, solver).await?;
+        let mut block_stream = into_stream(eth.current_block().clone());
+        loop {
+            // Wait for the next block to be mined or we time out. Block stream immediately
+            // yields the latest block, thus the first iteration starts immediately.
+            if tokio::time::timeout_at(deadline, block_stream.next())
+                .await
+                .is_err()
+            {
+                tracing::info!(?hash, "tx not confirmed in time, cancelling");
+                self.cancel(mempool, settlement.gas.price, solver).await?;
+                return Err(Error::Expired);
+            }
+            tracing::debug!(?hash, "checking if tx is confirmed");
+
+            let receipt = eth.transaction_receipt(&hash).await.unwrap_or_else(|err| {
+                tracing::warn!(?hash, ?err, "failed to get transaction receipt",);
```

Comment:
`warn` maybe too strict

Reply:
I think if we fail to request the status from the node that is unexpected (and should result in a warning)